sources/packages/backend/apps/test-db-seeding/src/db-seeding/02-institution/05-manage-designations.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { DesignationAgreement, InstitutionLocation } from "@sims/sims-db";
+import {
+  createFakeDesignationAgreement,
+  createMultipleFakeInstitutionLocations,
+} from "@sims/test-utils";
+import { Repository } from "typeorm";
+import {
+  DataSeed,
+  DataSeedMethod,
+  SeedPriorityOrder,
+} from "../../seed-executors";
+import { InstitutionHelperService } from "../../services";
+import { SIMS_COLL_F_LEGAL_SIGNING_USER } from "../constants";
+
+@Injectable()
+@DataSeed({ order: SeedPriorityOrder.Unknown })
+export class DesignationAgreementService {
+  constructor(
+    @InjectRepository(DesignationAgreement)
+    private readonly designationAgreementRepo: Repository<DesignationAgreement>,
+    @InjectRepository(InstitutionLocation)
+    private readonly institutionLocationRepo: Repository<InstitutionLocation>,
+    private readonly institutionHelperService: InstitutionHelperService,
+  ) {}
+
+  /**
+   * Method to seed fake pending designation agreement.
+   */
+  @DataSeedMethod()
+  async createPendingDesignationAgreement(): Promise<void> {
+    // Get fake institution.
+    const fakeInstitution =
```

Comment:
Asking for understanding, I see a number of creates here and should they go with transaction?